.github/workflows/check_verilog.yml

```diff
@@ -0,0 +1,117 @@
+name: synth-sv
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+env:
+  CIRCT_VERSION: 1.72.0
+  CIRCT_ARCHIVE: "circt-full-shared-linux-x64.tar.gz"
+  DEPENDENCIES_DIR: "deps"
+  MLIR_TARGETS_PATH: "lib/cmake/mlir/MLIRTargets.cmake"
+  BUILD_DIR: "build"
+  UTOPIA_EXECUTABLE: "src/umain"
+
+jobs:
+  synthesize-system-verilog:
+    strategy:
+      matrix:
+        kernel: [addconst, matrixmul2, movingsum, muxmul, polynomial2, scalar3]
+        config: [STUB]
+        ## A workaround to intiailize an empty config array.
+        exclude:
+          - config: STUB
+        include:
+          ## addconst configs.
+          - kernel: addconst
+            config: add_int_2
+          ## matrixmul configs.
+          - kernel: matrixmul2
+            config: add_int_2_mul_int3
+          ## movingsum configs.
+          - kernel: movingsum
+            config: add_int_2
+          - kernel: movingsum
+            config: add_int_8
```

Comment:
Is it a good practice to enumerate  different configs for same kernel in such way? Can't we do something like:
```
- kernel: somekernel
  config: [config1, config2]
```
?